.github/workflows/build-test-pr.yaml

```diff
@@ -0,0 +1,41 @@
+name: pr-test
+on:
+  pull_request:
+    branches:
+      - develop
+jobs:
+  docker-test:
+    runs-on: ubuntu-20.04
+    env:
+      DATABASE_NAME: docker
+      DATABASE_USERNAME: docker
+      DATABASE_PASSWORD: gis
+      DATABASE_HOST: db
+      DJANGO_SETTINGS_MODULE: core.settings.test_docker
+      VIRTUAL_HOST: projecta.kartoza.com
+      VIRTUAL_PORT: 8080
+      DUMPPREFIX: PG_projecta
+      PGUSER: docker
+      PGPASSWORD: gis
+      PGPORT: 5432
+      PGHOST: db
+      PGDATABASE: docker
+      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build containers
+        run: docker-compose --file deployment/docker-compose.test.yml build
+      - name: Run db container
+        run: docker-compose --file deployment/docker-compose.test.yml up -d db
+      - name: Run devweb container
+        run: docker-compose --file deployment/docker-compose.test.yml up --no-deps -d devweb
```

Comment:
You can reuse the docker build in action in https://github.com/kartoza/prj.app/blob/develop/.github/workflows/build-push-images-latest.yaml .

Reply:
Does it mean, I can edit the docker-file and using this image instead 
`${{ secrets.DOCKERHUB_REPO }}/projecta-uwsgi:latest` 
and I pass the secret value to env to be used by docker-compose.test ?

Reply:
Ideally the deployment should be configurable from just a config file or environment/parameters, so that the deployment code can be reused.

The plan is like this:
- Specify files/environment/parameters for running unittests
- Use docker build action to build image
- Use that image for the unittest
- If succeed, rebuild the image from cache, but this time, uses proper docker tag and then push to image registry.

To answer your question:

> Does it mean, I can edit the docker-file and using this image instead
`${{ secrets.DOCKERHUB_REPO }}/projecta-uwsgi:latest`
and I pass the secret value to env to be used by docker-compose.test ?

Use the docker build action to build the image. Since this is for test (we won't push it), you can just replace `${{ secrets.DOCKERHUB_REPO }}/projecta-uwsgi:latest` with hardcoded name, or Github Action env parameter. For example: `kartoza/projecta-uwsgi:test`. There is no need to pass the secret value to docker-compose.test since the image is already been built and available in the machine. You just need to use the matching tag in `docker-compose.test.yml`. 

Also, it's better to override instead of using `docker-compose -f docker-compose.test.yml`. Copy `docker-compose.test.yml` as `docker-compose.override.yml`. If you run `docker-compose up -d` it will automatically merged the yaml files of both `docker-compose.yml` and `docker-compose.override.yml`.